From 5a16c363749c11b1b0bf8693a395fa9d7749505c Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:02:29 -0400
Subject: [fs] btrfs: remove unnecessary inode generation lookup in send

Message-id: <1407179109-13714-263-git-send-email-sandeen@redhat.com>
Patchwork-id: 87149
O-Subject: [RHEL7.1 PATCH 262/422] Btrfs: remove unnecessary inode generation lookup in send
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Filipe Manana <fdmanana@gmail.com>

commit 425b5dafc8738d3d6d6b05827f40bd32bf04a20b
Author: Filipe Manana <fdmanana@gmail.com>
Date:   Tue Mar 18 17:56:06 2014 +0000

    Btrfs: remove unnecessary inode generation lookup in send

    No need to search in the send tree for the generation number of the inode,
    we already have it in the recorded_ref structure passed to us.

    Signed-off-by: Filipe David Borba Manana <fdmanana@gmail.com>
    Reviewed-by: Liu Bo <bo.li.liu@oracle.com>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/send.c b/fs/btrfs/send.c
index 6463691..92d4ae8 100644
--- a/fs/btrfs/send.c
+++ b/fs/btrfs/send.c
@@ -3179,7 +3179,7 @@ static int wait_for_parent_move(struct send_ctx *sctx,
  int ret;
  u64 ino = parent_ref->dir;
  u64 parent_ino_before, parent_ino_after;
- u64 new_gen, old_gen;
+ u64 old_gen;
  struct fs_path *path_before = NULL;
  struct fs_path *path_after = NULL;
  int len1, len2;
@@ -3197,12 +3197,7 @@ static int wait_for_parent_move(struct send_ctx *sctx,
  else if (ret < 0)
   return ret;
 
- ret = get_inode_info(sctx->send_root, ino, NULL, &new_gen,
-        NULL, NULL, NULL, NULL);
- if (ret < 0)
-  return ret;
-
- if (new_gen != old_gen)
+ if (parent_ref->dir_gen != old_gen)
   return 0;
 
  path_before = fs_path_alloc();
-- 
1.7.1